> 原文：<https://www.studytonight.com/kotlin/kotlin-exception-handling>

[← Prev](/kotlin/kotlin-extension-function "Kotlin Extension Function")[Next →](/kotlin/kotlin-try-catch "Kotlin try & catch")

# 科特林异常处理

在本教程中，我们将学习科特林中的**异常处理。几乎所有的编程语言都提供了处理异常的机制。在开始异常处理之前，让我们先讨论什么是异常。**

## **什么是例外？**

异常是中断程序正常流程的事件(或错误)。异常在运行时发生，并突然终止程序。异常可能由于许多原因而发生，例如:

*   **无效输入:**

    *   具有无效值的字符串

    *   空输入

    *   超出范围的输入数量

*   **编程错误:**

    *   超出数组索引范围

    *   无效操作，如被零除

*   **系统错误:**

    *   被遗忘

## **什么是异常处理？**

异常处理是处理运行时可能发生的异常的技术。如果我们想使我们的程序健壮，那么异常必须被处理。异常处理**使我们的程序免于突然终止**，并使我们能够向最终用户提供有意义的响应，解释异常发生的原因，而不是向他们显示异常消息(以及 strack trace)本身。

Kotlin 中的所有异常类都是类**可投掷**的后代。

**在 Java** 中，有两种类型的异常:

*   **选中异常:** [选中异常](https://www.studytonight.com/java/exception-handling.php)在编译时被选中。这些异常必须使用 try/catch 块来处理，或者通过在方法定义中提到`throws`关键字来进一步抛出异常。如果我们没有处理检查过的异常，程序就不会运行。

*   **未检查的异常:**编译时不检查这些异常。即使我们不处理这些异常，程序也会编译成功。

在科特林，我们只有**未检查的异常**，即**检查的异常在科特林**中不存在。消除检查异常背后的原因之一是，很大一部分开发人员认为检查异常会增加不必要的代码。

如果我们想让调用者知道函数可能抛出的异常，我们可以在 Kotlin 中添加`@Throws`注释。

## 柯特林中的异常处理

Kotlin 中的异常处理可以用四个关键词来理解。它们是:

1.  **Try** :可能发生异常的代码块被放在 Try 块中。try 块后面必须至少有一个 catch 块或一个 finally 块，或者两者都有。

2.  **接住**:放在试块后。它捕获由 try 块中的代码引发的异常。这是实际处理异常的块。我们可以在一个试块之后有多个 catch 块。

3.  **最后**:即使没有捕捉到异常或者根本没有发生异常，也始终执行 Finally block。Finally block 用于执行一些重要的操作，如关闭打开的文件。它被放置在 try 或 catch 块之后。最后只能有一个街区。

4.  **Throw** : Throw 关键字用于显式抛出异常。如果开发人员想在某些情况下抛出异常，我们可以使用 throw 关键字。

在接下来的教程中，我们将详细介绍它们以及代码示例。

## **总结**

在本教程中，我们了解了什么是异常，以及 Java 和 Kotlin 中异常类型的区别。在下一个教程中，我们将详细介绍如何使用`try`、`catch`、`finally`和`throw`关键词来处理科特灵中的异常。

* * *

* * *